From: olh@suse.de
Subject: Bug 374309 - EFIKA freezes during installation
Patch-mainline: never


Add some delay to the hot codepaths to avoid a hang during installation.
The memory barriers alone did not help, but an delay of 123 or 42 did help
to get past the hang after the vgscan --mknodes.


---
 arch/powerpc/platforms/52xx/mpc52xx_pic.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/powerpc/platforms/52xx/mpc52xx_pic.c
+++ b/arch/powerpc/platforms/52xx/mpc52xx_pic.c
@@ -52,6 +52,8 @@ static unsigned char mpc52xx_map_senses[
 	IRQ_TYPE_LEVEL_LOW,
 };
 
+#undef pr_debug
+#define pr_debug(x,y ...) do { mb(); eieio(); isync(); udelay(42); } while(0)
 /*
  *
 */
